src/go/k8s/apis/redpanda/v1alpha1/cluster_types.go

```diff
@@ -847,6 +866,30 @@ func (s *ClusterStatus) SetRestarting(restarting bool) {
 	s.DeprecatedUpgrading = restarting
 }
 
+// GetDesiredReplicas returns the current number of replicas that the controller wants to run.
+// It returns 1 when not initialized (as fresh clusters start from 1 replica)
+func (r *Cluster) GetDesiredReplicas() int32 {
+	if r.Status.DesiredReplicas <= 0 {
+		// Not initialized, let's give the computed value
+		return r.ComputeInitialDesiredReplicas()
+	}
+	return r.Status.DesiredReplicas
+}
+
+// ComputeInitialDesiredReplicas calculates the initial value for status.desiredReplicas.
+//
+// It needs to consider the following cases:
+// - Fresh cluster: we start from 1 replicas, then upscale if needed
```

Comment:
Checking HPA docs, I wonder whether this field should not be named currentreplicas rather? I feel like desired is the target and final one 🤔 https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/

but feel free to correct me, I just want to get the API right so it's not confusing since it's hard to change

Reply:
Yeah, I took the name from HPA, but I agree it's confusing, especially on the part where the user "desires" multiple replicas, but the controller keeps them to a lower value. `currentReplicas` may be better..

Reply:
The reason for 1 replica is to prevent the problem with seeds_servers, to have them empty only at cluster creation and never again